<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>WorkpieceIdentificationClass</class>
 <widget class="QMainWindow" name="WorkpieceIdentificationClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1193</width>
    <height>679</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>WorkpieceIdentification</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>271</width>
      <height>191</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="1">
      <widget class="QPushButton" name="pushButton_openCamera">
       <property name="text">
        <string>Open Camera</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QPushButton" name="pushButton_startDetection">
       <property name="text">
        <string>Start Detection</string>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QPushButton" name="pushButton_exit">
       <property name="text">
        <string>Exit</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="pushButton_reloadCamera">
       <property name="text">
        <string>Reload Camera</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QPushButton" name="pushButton_balanceWhite">
       <property name="text">
        <string>Balance White</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QPushButton" name="pushButton_savePicture">
       <property name="text">
        <string>Save Picture</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="pushButton_startSnap">
       <property name="text">
        <string>Start Snap</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QPushButton" name="pushButton_stopSnap">
       <property name="text">
        <string>Stop Snap</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>330</y>
      <width>281</width>
      <height>321</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="4" column="0" colspan="2">
      <widget class="QRadioButton" name="radioButton_unevenLight">
       <property name="text">
        <string>Uneven Light Compensation</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Exposure:100000ms</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Close Op K size:5</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QSlider" name="horizontalSlider_closeKSize">
       <property name="minimumSize">
        <size>
         <width>112</width>
         <height>20</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>112</width>
         <height>20</height>
        </size>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Other Param 2</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Canny Thresh A:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Other Param 1</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QSlider" name="horizontalSlider_exposureTime">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Canny Thresh B:</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Not Implemented Yet</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QSlider" name="horizontalSlider_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="spinBox_cannyA"/>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="spinBox_cannyB"/>
     </item>
     <item row="2" column="1">
      <widget class="QSpinBox" name="spinBox_3"/>
     </item>
     <item row="3" column="1">
      <widget class="QSpinBox" name="spinBox_4"/>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_display">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>10</y>
      <width>641</width>
      <height>641</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: 2px solid #000000; </string>
    </property>
    <property name="text">
     <string>  Camera Is Not Opened Yet</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>10</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt;&quot;&gt;Detection Result:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_showResult">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>10</y>
      <width>231</width>
      <height>641</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: 2px solid #000000; </string>
    </property>
    <property name="text">
     <string>Camera Is Not Opened Yet</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_cameraInfo">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>230</y>
      <width>271</width>
      <height>81</height>
     </rect>
    </property>
    <property name="text">
     <string>Camera Succeddfully Loaded.
Model:
MAC:</string>
    </property>
   </widget>
   <zorder>gridLayoutWidget</zorder>
   <zorder>gridLayoutWidget_2</zorder>
   <zorder>label_display</zorder>
   <zorder>label_showResult</zorder>
   <zorder>label_9</zorder>
   <zorder>label_cameraInfo</zorder>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="WorkpieceIdentification.qrc"/>
 </resources>
 <connections/>
 <slots>
  <slot>slot_pushButton_openCamera()</slot>
  <slot>slot_pushButton_balanceWhite()</slot>
  <slot>slot_pushButton_savePicture()</slot>
  <slot>slot_pushButton_exit()</slot>
  <slot>slot_pushButton_startDetection()</slot>
  <slot>slot_lineEdit_cannyA()</slot>
  <slot>slot_lineEdit_cannyB()</slot>
  <slot>slot_radioButton_unevenLight()</slot>
  <slot>slot_slider_closeKSize()</slot>
  <slot>slot_slider_exposureTime()</slot>
 </slots>
</ui>
